src/unsatcores/MinUnsatCoreBuilder.cc

```diff
@@ -0,0 +1,82 @@
+
+#include "MinUnsatCoreBuilder.h"
+
+#include <api/MainSolver.h>
+
+namespace opensmt {
+
+std::unique_ptr<UnsatCore> MinUnsatCoreBuilder::build() {
+    buildBody();
+    return buildReturn();
+}
+
+void MinUnsatCoreBuilder::buildBody() {
+    UnsatCoreBuilder::buildBody();
+    minimize();
+}
+
+void MinUnsatCoreBuilder::minimize() {
+    minimizeInit();
+    minimizeAlg();
+    minimizeFinish();
+}
+
+void MinUnsatCoreBuilder::minimizeInit() {
+    assert(terms.size() >= namedTerms.size());
+    assert(size_t(namedTerms.size()) == namedTermsIdxs.size());
+}
+
+void MinUnsatCoreBuilder::minimizeAlgNaive() {
```

Comment:
This algorithm only minimizes the named terms, no?
Thus this does not compute minimal explanation if used together with the option `print-cores-full`, no?

Reply:
My intuition is that if `print-cores-full` is set then the names should be ignored and, if minimal cores are required then the minimization algorithm should operate on all terms, not only the named ones.

Reply:
-> #770